os/kernel/log_dump/log_dump.c

```diff
@@ -109,10 +137,79 @@ int log_dump_init(void)
 	return LOG_DUMP_OK;
 }
 
-void log_dump_read_init(void)
+unsigned char *last_comp_block;
+size_t last_comp_block_size;
+int last_comp_block_ptr;
+bool last_block_compression;
+
+/* This is used to wake up the compression thread to compress current log data */
+int log_dump_read_wake(void)
+{
+	last_block_compression = true;
+	struct log_dump_msg_s msg;
+	int ret;
+	mqd_t mq_fd;
+
+	/* Open log dump message queue */
+	mq_fd = mq_open(LOG_DUMP_MSGQ, O_WRONLY, 0666, 0);
+	if (mq_fd == (mqd_t)ERROR) {
+		lldbg("Fail to open mq, errno %d. EXIT!\n", errno);
+		return ERROR;
+	}
+
+	msg.comp = true;
+	ret = mq_send(mq_fd, (const char *)&msg, sizeof(struct log_dump_msg_s), 100);
+	if (ret < 0) {
+		mq_close(mq_fd);
+		lldbg("failed mq_send ret = %d\n", ret);
+		return ERROR;
+	}
+
+	mq_close(mq_fd);
+	while (last_block_compression) ;
```

Comment:
Why do you wait `last_block_compression`?

Reply:
It is used to synchronize completion of last partially filled block compression